### ISSUE TITLE
New patch for sms.cpp: more accurate FM Sound Unit behavior

### ISSUE THREAD

**Issue by enikland2** [mamedev/mame#1348]
Micko asked me to post a diff here as an alternative for patch submission.
Changelog:

- more accurate FM Sound Unit behavior, based on Charles' analysis of Enri's schematics;
- fix order of media slots when the gender adapter is plugged to the expansion slot of a Japanese/Korean SMS driver.

--
Enik

Edit (2016-09-03): Two lines modified by the original patch were reverted.
Edit (2016-09-05): Add Pit Pot as an example of conflict with the FM unit.

[mame_sms_diff.txt](https://github.com/mamedev/mame/files/455703/mame_sms_diff.txt)

**Comment by p1pkin** [comment]
why not regular pull request ?

**Comment by enikland2** [comment]
I tried the regular method before posting this patch as an issue. The procedure doesn't seem difficult, but I noticed it would take more than 900MB just to clone the repository, what is impracticable with my limited connection. The bz2 file I downloaded from the git web interface had 85MB. Sorry if contribute to MAME requires high end PCs and internet connection, but my situation at least prevents that I waste my time with something else like modern PC games.
